src/app/PriceChart.tsx

```diff
@@ -0,0 +1,265 @@
+import { createChart, CandlestickData, UTCTimestamp } from "lightweight-charts";
+import React, { useEffect, useRef, useContext, useState } from "react";
+import { AdexStateContext } from "./page";
+import * as adex from "alphadex-sdk-js";
+
+// from https://www.npmjs.com/package/alphadex-sdk-js#candleperiods-array
+const CANDLE_PERIODS = ["5m", "15m", "1h", "4h", "6h", "12h", "1D", "1W", "1M"];
+
+async function fetchHistoricalData(
+  symbol: string,
+  interval: string,
+  limit: number
+): Promise<OHLCVData[]> {
+  const response = await fetch(
+    `https://api.binance.com/api/v3/klines?symbol=${symbol}&interval=${interval}&limit=${limit}`
+  );
+  const data = await response.json();
+
+  return data.map((row: any): OHLCVData => {
+    const [time, open, high, low, close, value] = row;
+    const tradingviewTime = (time / 1000) as UTCTimestamp;
+    return {
+      time: tradingviewTime,
+      open: Number(open),
+      high: Number(high),
+      low: Number(low),
+      close: Number(close),
+      value: Number(value),
+    };
+  });
+}
+
+interface OHLCVData extends CandlestickData {
+  value: number;
+}
+
+interface PriceChartProps {
+  data: OHLCVData[];
+}
+
+enum DataSource {
+  BINANCE = "BINANCE",
+  ADEX = "ADEX",
+}
+
+function PriceChartCanvas(props: PriceChartProps) {
+  const chartContainerRef = useRef<HTMLDivElement>(null);
+  const { data } = props;
```

Comment:
Nice solution as it will only sort again if necessary. I am planning to refactor the SDK quite extensively once RCNet2 is up and running and will put this on the list to investigate. But it is very weird that the same code will result in different lists based on the browser. Just to check, have you checked any other lists from Adex to see if they are also reversed in firefox. I will definitely include this in my own tests for the next version.

Reply:
yeah, but it doesn't guarantee that that the array will be sorted, what if the input array for whatever reason was not just reversed but shuffled (by this time I won't be surprised if say safari does something like this 😆 )